cmd/permissions-checker/permissionschecks.go

```diff
@@ -0,0 +1,114 @@
+					printOutput(level, "d", fullPath, currentPerm, DefaultDirPerm, currentOwner.Username, currentGroup.Name, hasCorrectPermissions, hasCorrectOwner, hasCorrectGroup)
+					FindRecursive(fullPath, level)
+				}
+			}
+		}
+	}
+}
+
+// checkPermissions return true if the permissions are correct (false in other case)
+/* currentPermissions [string] - permissions that the file or directory has (i.e. -rwxrwxr-x)
+ * defaultPermissions [string] - permissions that should have (i.e. rwxrwxr-x)
+ */
+func checkPermissions(currentPermissions, defaultPermissions string) bool {
+	return strings.Contains(currentPermissions, defaultPermissions)
+}
+
+// printOutput Print the data in different formats according to the situation
+/* level [string] 							- used to print the output in a hierarchical way
+ * kind [string] 								- "f" if it is file or "d" in case of a directory
+ * name [string] 				  			- receive the full path of the file/directory (if verbose only use the name)
+ * currentPermissions [string] 	- file/directory current permissions
+ * defaultPermissions [string] 	- file/directory default permissions
+ * currentOwner [string] 				- file/directory current owner
+ * currentGroup [string] 				- file/directory current group
+ * hasCorrectPermissions [bool] - true if the permissions are correct
+ * hasCorrectOwner [bool] 			- true if the owner is correct
+ * hasCorrectGroup [bool] 			- true if the group is correct
+ */
+func printOutput(level int, kind, name, currentPermissions, defaultPermissions, currentOwner, currentGroup string, hasCorrectPermissions, hasCorrectOwner, hasCorrectGroup bool) {
+	if Verbose {
+		name = filepath.Base(name)
+	}
+
+	if hasCorrectPermissions && hasCorrectOwner && hasCorrectGroup { // Everything correct
+		if Verbose {
+			fmt.Printf("%s(%s) %s %s %s %s\n", strings.Repeat(" ", level), kind, name, currentPermissions, currentOwner, currentGroup)
+		}
+	} else if hasCorrectPermissions && (!hasCorrectOwner || !hasCorrectGroup) { // Permissions correct, fails owner or group
+		fmt.Printf("\x1b[31;1m%s(%s) %s %s %s %s (expected %s %s)\n\x1b[0m", strings.Repeat(" ", level), kind, name, currentPermissions, currentOwner, currentGroup, DefaultOwner, DefaultGroup)
+	} else if !hasCorrectPermissions && (hasCorrectOwner && hasCorrectGroup) { // Permissions wrong, owner and group correct
+		fmt.Printf("\x1b[31;1m%s(%s) %s %s (expected %s) %s %s\n\x1b[0m", strings.Repeat(" ", level), kind, name, currentPermissions, defaultPermissions, currentOwner, currentGroup)
+	} else if !hasCorrectPermissions && (!hasCorrectOwner || !hasCorrectGroup) { // Nothing correct
+		fmt.Printf("\x1b[31;1m%s(%s) %s %s (expected %s) %s %s (expected %s %s)\n\x1b[0m", strings.Repeat(" ", level), kind, name, currentPermissions, defaultPermissions, currentOwner, currentGroup, DefaultOwner, DefaultGroup)
+	}
+}
+
+// ManageInputs Validate and/or format the parameters entered by the user
+func ManageInputs() {
```

Comment:
This would make sense to be in the main.go file, it is not related to the tests. What is more, I am not sure if this refactor would make sense

Reply:
It has sense because if not when the output is printed you could have lines with double `//`
```
(f) /Applications/wordpress-4.9.4-6/apache2//scripts/ctl.sh -rwxr-xr-x (expected -rw-r--r--) crhernandez admin
```
instead of 
```
(f) /Applications/wordpress-4.9.4-6/apache2/scripts/ctl.sh -rwxr-xr-x (expected -rw-r--r--) crhernandez admin
```

In this way, I unified the criteria regardless of whether the user uses a path ending on `/` or not.